proxy/proxy.go

```diff
@@ -119,6 +132,31 @@ func (proxy *Proxy) ListenAndServe() {
 	}
 }
 
+func unescapeBackSlashes(src string) string {
+	return escapedSlashRegexp.ReplaceAllString(src, "/")
+}
+
+func parseHostname(hostname string) (*regexp.Regexp, string, error) {
+	namedSubmatches := findStringNamedSubmatch(substitutionRegexp, hostname)
+	if len(namedSubmatches) < 1 {
+		errStr := fmt.Sprintf("Incorrect hostname expression: %s", hostname)
+		return nil, "", errors.New(errStr)
+	}
+
+	escapedPattern := namedSubmatches[substitutionPatternName]
+	escapedReplacement := namedSubmatches[substitutionReplacementName]
+	unescapedPattern := unescapeBackSlashes(escapedPattern)
+	unescapedReplacement := unescapeBackSlashes(escapedReplacement)
```

Comment:
I, at the very least,  need `escapedPattern` to report an error if `unescapedPattern` doesn't compile (read below).